src/scroll-timeline-css-parser.js

```diff
@@ -0,0 +1,327 @@
+    // TODO is it enough to check with "includes()"
+    const hasAnimationName = rule.block.contents.includes("animation-name:");
+    const hasScrollTimeline = rule.block.contents.includes("animation-timeline:");
+    const hasAnimation = rule.block.contents.includes("animation:");
+
+    // If both 'animation-timeline' and 'animation-name' are present,
+    // save them in the list
+    if (hasScrollTimeline && hasAnimationName) {
+      let timelineNames = /animation-timeline\s*:([^;}]+)/
+        .exec(rule.block.contents)?.[1]
+        .trim().split(",").map(name => name.trim());
+
+      let animationNames = /animation-name\s*:([^;}]+)/
+        .exec(rule.block.contents)?.[1]
+        .trim().split(",").map(name => name.trim());
+
+      for (let i = 0; i < timelineNames.length; i++) {
+        this.cssRulesWithTimelineName.push({
+          selector: rule.selector,
+          'animation-name': animationNames[i],
+          'animation-timeline': timelineNames[i]
+        });
+      }
+      return;
+    }
+
+    let timelineName = /animation-timeline\s*:([^;}]+)/
+      .exec(rule.block.contents)?.[1]
+      .trim();
+
+    let animationName = undefined;
+    if (hasAnimationName) {
+      animationName = /animation-name\s*:([^;}]+)/
+        .exec(rule.block.contents)?.[1]
+        .trim();
+    } else if (hasAnimation) {
+      let shorthand = /animation\s*:([^;}]+)/
+        .exec(rule.block.contents)?.[1]
+        .trim();
+
+      if (shorthand) {
+        let remainingTokens = removeKeywordsFromAnimationShorthand(shorthand);
+        // TODO we are assuming the first one that is remaining is
+        // definitely animation name, and the second one is definitely
+        // scrollTimeline name, which may not be true!
+        if (remainingTokens.length <= 2)
+          animationName = remainingTokens[0];
+
+        if (remainingTokens.length == 2) {
+          timelineName = remainingTokens[1];
+          // Remove timeline name from animation shorthand
+          // so the native implementation works with the rest of the properties
+          rule.block.contents = rule.block.contents.replace(
+            timelineName,
+            ""
+          );
+          this.replacePart(
+            rule.block.startIndex,
+            rule.block.endIndex,
+            rule.block.contents,
+            p
+          );
+        }
+      }
+    }
+
+    if (animationName && timelineName) {
+      this.cssRulesWithTimelineName.push({
+        selector: rule.selector,
+        'animation-name': animationName,
+        'animation-timeline': timelineName
+      });
+    }
+
+    // The animation-timeline property may not be used in keyframes
+    if (timelineName && !rule.selector.includes("@keyframes")) {
+      this.cssRulesWithTimelineName.push({
+        selector: rule.selector,
+        'animation-name': undefined,
+        'animation-timeline': timelineName
+      });
+    }
+  }
+
+  parseIdentifier(p) {
```

Comment:
Did his parse code come from another project?  If so, can you cite the project.  Is the parse code bundled in such a manner than it can be reused without copying?